### PR TITLE
Back port changes in JITClientCompilationThread from master

### PR DESCRIPTION
**Pull request by a7ehuo**
Include PR #7215 (`JITClientCompilationThread`) and PR #7151(`j9methodServer`) from the `master` branch.

Signed-off-by: Annabelle Huo <Annabelle.Huo@ibm.com>

### CODE REVIEW

**Comment by a7ehuo** [comment]
This PR has dependency __#7237__ (Update __JITServerRelocationRuntime__).

File: runtime/compiler/control/CompilationRuntime.hpp [hunk 1]
```diff
@@ -827,8 +827,9 @@ class CompilationInfo
    void     setAotRelocationTime(uint32_t reloTime) { _statTotalAotRelocationTime = reloTime; }
    void    incrementNumMethodsFoundInSharedCache() { _numMethodsFoundInSharedCache++; }
    int32_t numMethodsFoundInSharedCache() { return _numMethodsFoundInSharedCache; }
+#if defined(JITSERVER_SUPPORT)
```
Comment (mpirvu) [inline runtime/compiler/control/CompilationRuntime.hpp]: Is this still needed? Maybe until Harry ports the changes from Younes

File: runtime/compiler/env/j9methodServer.cpp [hunk 2]
```diff
@@ -2245,6 +2246,13 @@ TR_ResolvedRelocatableJ9JITServerMethod::isUnresolvedMethodHandle(I_32 cpIndex)
    return true;
    }
 
+char *
```
Comment (mpirvu) [inline runtime/compiler/env/j9methodServer.cpp]: This method is something I removed from jitaas and should be removed from master branch as well.

File: runtime/compiler/env/j9methodServer.hpp [hunk 3]
```diff
@@ -274,6 +301,7 @@ class TR_ResolvedRelocatableJ9JITServerMethod : public TR_ResolvedJ9JITServerMet
    virtual void                  handleUnresolvedStaticMethodInCP(int32_t cpIndex, bool * unresolvedInCP) override;
    virtual void                  handleUnresolvedSpecialMethodInCP(int32_t cpIndex, bool * unresolvedInCP) override;
    virtual void                  handleUnresolvedVirtualMethodInCP(int32_t cpIndex, bool * unresolvedInCP) override;
+   virtual char *                fieldOrStaticNameChars(int32_t cpIndex, int32_t & len) override;
```
Comment (mpirvu) [inline runtime/compiler/env/j9methodServer.hpp]: This method is something I removed from jitaas and should be removed from master branch as well.

File: runtime/compiler/control/CompilationRuntime.hpp (hunk 1 above)
Comment (a7ehuo) [reply runtime/compiler/control/CompilationRuntime.hpp]: If this PR is merged before #7344, `isRomClassForMethodInSharedCache` should be kept. #7344 should rebase first. If this PR is merged after, I'll rebase this PR to the latest `jitaas` and remove `isRomClassForMethodInSharedCache`.

File: runtime/compiler/env/j9methodServer.hpp (hunk 3 above)
Comment (a7ehuo) [reply runtime/compiler/env/j9methodServer.hpp]: Will remove it. I thought it was the other way around.

**Comment by a7ehuo** [comment]
@mpirvu All comments addressed. Ready for another review. Thanks.

**Comment by mpirvu** [comment]
Jenkins test sanity.functional+jitaas xlinux jdk8,jdk11 depends eclipse/openj9-omr#jitaas

**Comment by mpirvu** [comment]
Only the known `cmdLineTester_gcsuballoctests_0` test failed, hence merging.

**Pull request merged**